BinaryEncoding.md

```diff
@@ -234,6 +233,52 @@ This section must be preceded by a [Functions](#functions-section) section.
 | count | `varuint32` | count of entries to follow |
 | entries | `uint16*` | repeated indexes into the function table |
 
+### Function Names section
+
+This section may occur 0 or 1 times and does not change execution semantics. A
+validation error in this section is not reported and is treated as the section
+being absent. The expectation is that, when a binary WebAssembly module is
+viewed in a browser or other development environment, the names in this section
+will be used as the names of functions in the [text format](TextFormat.md).
+
+ID: `function_names`
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| names | `name*` | sequence of names |
+
+The number of names is determined by the number of declared functions. The
+sequence of names assigns a name to each function index. A name is:
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| size | varuint32 | string length, in bytes |
+| string | bytes | valid utf8 encoding |
```

Comment:
The practical reason is that, if we don't mandate utf8, then it's possible tools will start generating ascii or utf16 or ... and I think that could tangibly hurt the wasm ecosystem.  It'd be html content-sniffing days all over again (except no MIME types or <meta> tags to help out).  Here, the purpose of "render to text" is baked into wasm; with import/export strings we're intentionally throwing up our hands and letting the host decide (and the contents might not even be text, but some structured data rep that, say, describes lookup parameters).